testing/functional_tests_assertions.rst

```diff
@@ -0,0 +1,79 @@
+.. index::
+   single: Tests; Assertions
+
+Functional Test specific assertions Reference
+=============================================
+
+When doing functional tests, sometimes you need to make complex assertions in
+order to check whether the ``Request``, the ``Response`` or the ``Crawler``
+contain the expected information to make your test succeed.
+
+Here is an example without the specific assertions::
+
+    $this->assertGreaterThan(
+        0,
+        $crawler->filter('html:contains("Hello World")')->count()
+    );
+
+Now here is the example with the assertions introduced in Symfony 4.3::
+
+    $this->assertSelectorTextContains('html', 'Hello World');
+
+As you can see, even though the first assertion uses native PHPUnit methods,
+the second one explains more what we want to expect.
+
+As of Symfony 4.3, a list of assertions have been added, here is the reference.
```

Comment:
I would not mention 4.3 specifically anywhere in this new doc, but instead add a `addedversion` section at the top of the doc.